dmutils/email/dm_mailchimp.py

```diff
@@ -0,0 +1,101 @@
+
+    def create_campaign(self, campaign_data):
+        try:
+            campaign = self.client.campaigns.create(campaign_data)
+            return campaign['id']
+        except RequestException as e:
+            self.logger.error(
+                "Mailchimp failed to create campaign for '{0}'".format(
+                    campaign_data.get("settings").get("title")
+                ),
+                extra={"error": str(e)}
+            )
+        return False
+
+    def set_campaign_content(self, campaign_id, content_data):
+        try:
+            return self.client.campaigns.content.update(campaign_id, content_data)
+        except RequestException as e:
+            self.logger.error(
+                "Mailchimp failed to set content for campaign id '{0}'".format(campaign_id),
+                extra={"error": str(e)}
+            )
+        return False
+
+    def send_campaign(self, campaign_id):
+        try:
+            self.client.campaigns.actions.send(campaign_id)
+            return True
+        except RequestException as e:
+            self.logger.error(
+                "Mailchimp failed to send campaign id '{0}'".format(campaign_id),
+                extra={"error": str(e)}
+            )
+        return False
+
+    def subscribe_new_email_to_list(self, list_id, email_address):
+        """ Will subscribe email address to list if they do not already exist in that list else do nothing"""
+        hashed_email = self.get_email_hash(email_address)
+        try:
+            return self.client.lists.members.create_or_update(
+                list_id,
+                hashed_email,
+                {
+                    "email_address": email_address,
+                    "status_if_new": "subscribed"
+                }
+            )
+        except RequestException as e:
+            # expected error due to user email being on Mailchimp blacklist. Don't fail but log.
+            if "looks fake or invalid, please enter a real email address." in e.response.json()["detail"]:
+                self.logger.error(
+                    "Expected error: Mailchimp failed to add user ({}) to list ({}). API error: The email address looks fake or invalid, please enter a real email address.".format(  # noqa
+                        hashed_email,
+                        list_id
+                    ),
+                    extra={"error": str(e)}
+                )
+                return True
+            self.logger.error(
+                "Mailchimp failed to add user ({}) to list ({})".format(
+                    hashed_email,
+                    list_id
+                ),
+                extra={"error": str(e)}
+            )
+            return False
+
+    def subscribe_new_emails_to_list(self, list_id, email_addresses):
+        success = True
+        for email_address in email_addresses:
+            if not self.subscribe_new_email_to_list(list_id, email_address):
+                success = False
+        return success
```

Comment:
This wouldn't work. Because you've used a generator expression `all` will yield at the first `False` causing only part of the emails to be added.
```
foo = 0

def bar1():
    global foo
    foo +=1

def bar2():
    global foo
    foo +=1
    return False

def bar3():
    global foo
    foo +=1

all(i() for i in [bar1, bar2, bar3])
print foo
>>> 1

foo = 0
[i() for i in [bar1, bar2, bar3]]
print foo
>>> 3
```

This would work:
```
return all([self.subscribe_new_email_to_list(list_id, email_address) for email_address in email_addresses])
```
But it's inadvisable to perform side effects like this in a comprehension anyway.
If this had been written like so:
```
for email_address in email_addresses:
     if not self.subscribe_new_email_to_list(list_id, email_address):
         yield False
```
this error would have been much easier to catch. As it stands it took a bit of pseudo code before I was fully confident of the behaviour here.